openapi3/schema_formats.go

```diff
@@ -21,26 +21,111 @@ type FormatCallback func(value string) error
 
 // Format represents a format validator registered by either DefineStringFormat or DefineStringFormatCallback
 type Format struct {
+	versionedFormats []*versionedFormat
+}
+
+type versionedFormat struct {
 	regexp   *regexp.Regexp
 	callback FormatCallback
 }
 
+func (format *Format) add(minMinorVersion uint64, vFormat *versionedFormat, override bool) {
+	if format != nil {
+		if format.versionedFormats == nil {
+			format.versionedFormats = make([]*versionedFormat, minMinorVersion+1)
+			format.versionedFormats[minMinorVersion] = vFormat
+		} else {
+			numVersionedFormats := uint64(len(format.versionedFormats))
+			if minMinorVersion >= numVersionedFormats {
+				// grow array
+				lastValue := format.versionedFormats[numVersionedFormats-1]
+				additionalEntries := make([]*versionedFormat, minMinorVersion+1-numVersionedFormats)
+				if lastValue != nil {
+					for i := 0; i < len(additionalEntries); i++ {
+						additionalEntries[i] = lastValue
+					}
+				}
+				format.versionedFormats = append(format.versionedFormats, additionalEntries...)
+				format.versionedFormats[minMinorVersion] = vFormat
+				return
+			}
+			if override {
+				for i := minMinorVersion; i < numVersionedFormats; i++ {
+					format.versionedFormats[i] = vFormat
+				}
+
+			} else {
+				minValue := format.versionedFormats[minMinorVersion]
+				for i := minMinorVersion; i < numVersionedFormats && minValue == format.versionedFormats[i]; i++ {
+					format.versionedFormats[i] = vFormat
+				}
+
+			}
+		}
+	}
+}
+
+func (format Format) get(minorVersion uint64) *versionedFormat {
+	if format.versionedFormats != nil {
+		if minorVersion >= uint64(len(format.versionedFormats)) {
+			return format.versionedFormats[len(format.versionedFormats)-1]
+		}
+		return format.versionedFormats[minorVersion]
+	}
+	return nil
+}
+
+func (format Format) DefinedForMinorVersion(minorVersion uint64) bool {
```

Comment:
Will have to determine if we keep it so one can determine if a definition already exists for a specific minor version. Since format does not expose its content, this is the only way for one to know about it.

If we keep, I will need to add tests